dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/DataPointQaReviewRepository.kt

```diff
@@ -0,0 +1,96 @@
+package org.dataland.datalandqaservice.org.dataland.datalandqaservice.repositories
+
+import org.dataland.datalandqaservice.org.dataland.datalandqaservice.entities.DataPointQaReviewEntity
+import org.dataland.datalandqaservice.org.dataland.datalandqaservice.utils.DataPointQaReviewItemFilter
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
+import java.util.UUID
+
+/**
+ * A JPA repository for accessing QA information of a data point
+ */
+interface DataPointQaReviewRepository : JpaRepository<DataPointQaReviewEntity, UUID> {
+    /**
+     * A function to get the dataId of the currently active data point, given a triple of company ID, data type and reporting period
+     * @param filter the filter to apply to the search containing the company ID, data type, reporting period and the QA status
+     */
+    @Query(
+        "SELECT dataPointQaReview.dataId FROM DataPointQaReviewEntity dataPointQaReview " +
+            "WHERE dataPointQaReview.companyId = ':#{#filter.companyId}' " +
+            "AND dataPointQaReview.dataPointIdentifier = ':#{#filter.dataPointIdentifier}' " +
+            "AND dataPointQaReview.reportingPeriod = ':#{#filter.reportingPeriod}' " +
+            "AND dataPointQaReview.qaStatus = 'Accepted' " +
+            "ORDER BY dataPointQaReview.timestamp DESC " +
+            "LIMIT 1",
+    )
+    fun getDataIdOfCurrentlyActiveDataPoint(
+        @Param("filter") filter: DataPointQaReviewItemFilter,
+    ): String?
```

Comment:
Die Query-Logik klappt glaube ich nicht (Wird nicht geprüft, ob qa-status nach accept rejected wurde). Außerdem ist der filter etwas grob: Warum hat der eine property für den status, wenn die in der query nicht genutzt wird.

Reply:
Ja das ist mir auch aufgefallen und es ist korrigiert